javascript-carousel/main.js

```diff
@@ -0,0 +1,107 @@
+  currentImage = parseInt(event.target.getAttribute('data-image-index'));
+  $img.setAttribute('src', imageSrcArray[parseInt(event.target.getAttribute('data-image-index'))]);
+  for (let i = 0; i < $circleSelectorAll.length; i++) {
+    if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) !== parseInt(event.target.getAttribute('data-image-index'))) {
+      $circleSelectorAll[i].className = 'circle';
+    }
+  }
+  timerId = setInterval(updatingPicture, 3000);
+}
+
+function lastPicture(event) {
+  clearInterval(timerId);
+  if (currentImage === 0) {
+    currentImage = imageSrcArray.length - 1;
+    $img.setAttribute('src', imageSrcArray[currentImage]);
+    $img.setAttribute('data-image-index', currentImage);
+    for (let i = 0; i < $circleSelectorAll.length; i++) {
+      if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+        $circleSelectorAll[i].className = 'circle circle-background';
+      } else {
+        $circleSelectorAll[i].className = 'circle';
+      }
+    }
+    timerId = setInterval(updatingPicture, 3000);
+    return;
+  }
+  currentImage = parseInt($img.getAttribute('data-image-index')) - 1;
+  $img.setAttribute('src', imageSrcArray[currentImage]);
+  $img.setAttribute('data-image-index', parseInt($img.getAttribute('data-image-index')) - 1);
+  for (let i = 0; i < $circleSelectorAll.length; i++) {
+    if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+      $circleSelectorAll[i].className = 'circle circle-background';
+    } else {
+      $circleSelectorAll[i].className = 'circle';
+    }
+  }
+  timerId = setInterval(updatingPicture, 3000);
+}
+
+function nextPicture(event) {
+  clearInterval(timerId);
+  if (currentImage === imageSrcArray.length - 1) {
+    currentImage = 0;
+    $img.setAttribute('src', imageSrcArray[currentImage]);
+    $img.setAttribute('data-image-index', currentImage);
+    for (let i = 0; i < $circleSelectorAll.length; i++) {
+      if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+        $circleSelectorAll[i].className = 'circle circle-background';
+      } else {
+        $circleSelectorAll[i].className = 'circle';
+      }
+    }
+    timerId = setInterval(updatingPicture, 3000);
+    return;
+  }
+  currentImage = parseInt($img.getAttribute('data-image-index')) + 1;
+  $img.setAttribute('src', imageSrcArray[currentImage]);
+  $img.setAttribute('data-image-index', parseInt($img.getAttribute('data-image-index')) + 1);
+  for (let i = 0; i < $circleSelectorAll.length; i++) {
+    if (parseInt($circleSelectorAll[i].getAttribute('data-image-index')) === parseInt($img.getAttribute('data-image-index'))) {
+      $circleSelectorAll[i].className = 'circle circle-background';
+    } else {
+      $circleSelectorAll[i].className = 'circle';
+    }
+  }
```

Comment:
This is pretty much the same code twice. If you update the current image and just roll with it, you don't need all of this code.